### DISCUSSION TITLE
my own language in nes console

### DISCUSSION THREAD

**Discussion by RodrigoDornelles** [cc65/cc65#1880] [Show and tell]
I'm building a tiny virtual machine that can run anywhere _(written in strict ansi c with minimal dependencies and lots of premature optimizations)_, and I intend to bring here my advances in the 6502 processor family!

 * official website <https://3bc-lang.org/>
 * repo source code <https://github.com/rodrigodornelles/3bc-lang/>
 
 ### 1- a small step ###

to some it doesn't seem like much, but I'm happy that the virtual machine actually fit in a cartridge.

there's still a lot ahead!

| ![image](https://user-images.githubusercontent.com/29123479/197288693-68d8a726-bd5f-49c8-aad4-8898c9ad213a.png) |
| :-: |

ignore that for now it overflows memory easily, but the data structures will be changed. :laughing:

| extra:<br/>virtual keyboard |
| :-: |
| ![Screen Shot 2022-10-21 at 18 11 56](https://user-images.githubusercontent.com/29123479/197289453-e2b1fa1e-fdb9-4a5c-992e-f1a0d1db8940.png) |

**Comment by thiswallz** [comment]
hey keep it up, nice
